src/pygama/lgdo/lh5_store.py

```diff
@@ -1,100 +1,116 @@
+"""
+This module implements routines from reading and writing LEGEND Data Objects in
+HDF5 files.
+"""
+from __future__ import annotations
+
 import fnmatch
 import glob
 import logging
 import os
 import sys
 from bisect import bisect_left, bisect_right
 from collections import defaultdict
+from typing import Any
 
 import h5py
 import numpy as np
 import pandas as pd
 
-from .array import Array
-from .arrayofequalsizedarrays import ArrayOfEqualSizedArrays
-from .fixedsizearray import FixedSizeArray
-from .lgdo_utils import *
-from .scalar import Scalar
-from .struct import Struct
-from .table import Table
-from .vectorofvectors import VectorOfVectors
-from .waveform_table import WaveformTable
+from pygama.lgdo.array import Array
+from pygama.lgdo.arrayofequalsizedarrays import ArrayOfEqualSizedArrays
+from pygama.lgdo.fixedsizearray import FixedSizeArray
+from pygama.lgdo.lgdo_utils import parse_datatype
+from pygama.lgdo.scalar import Scalar
+from pygama.lgdo.struct import Struct
+from pygama.lgdo.table import Table
+from pygama.lgdo.vectorofvectors import VectorOfVectors
+from pygama.lgdo.waveform_table import WaveformTable
+
+LGDO = Array | Scalar | Struct | VectorOfVectors
```

Comment:
I define here this type alias representing LGDOs. I include base classes only because that should be enough.